lib/class-core.php

```diff
@@ -322,8 +366,53 @@ public function admin_redirect() {
 
 		$site = $this->get_default_site();
 		if ( ! empty( $site->name ) && false === strpos( $_SERVER['HTTP_HOST'], $site->name ) ) { // WPCS: sanitization ok.
-			wp_redirect( esc_url_raw( 'http://' . $site->name . $_SERVER['REQUEST_URI'] ) ); // WPCS: sanitization ok.
+			wp_safe_redirect( esc_url_raw( sprintf( 'http%s://%s%s', is_ssl() ? 's' : '', $site->name, $_SERVER['REQUEST_URI'] ) ) ); // WPCS: sanitization ok.
 			exit();
 		}
 	}
+
+	/**
+	 * Filter the allowed redirect hosts used by wp_safe_redirect to add
+	 * Switchboard domains.
+	 *
+	 * @param  array $hosts Allowed hosts.
+	 * @return array
+	 */
+	public function allowed_redirect_hosts( $hosts ) {
+		$domains = get_option( 'switchboard_sites', [] );
+		if ( ! empty( $domains ) ) {
+			$domains = array_keys( $domains );
+			$hosts = array_merge( $hosts, $domains );
+		}
+		return $hosts;
+	}
+
+	/**
+	 * Get domain aliases as an array of alias => domain.
+	 *
+	 * @return array Keys are aliases, values are the aliased domains. For
+	 *               example, [ 'alias.domain.com' => 'domain.com' ].
+	 */
+	public static function get_domain_aliases() {
+		$cache_key = 'switchboard-domain-aliases';
+		$aliases = get_transient( $cache_key );
+		if ( false === $aliases ) {
+			$aliases = [];
+			$domains = get_option( 'switchboard_sites', [] );
+			foreach ( $domains as $domain => $domain_props ) {
+				$domain_aliases = get_term_meta( $domain_props['term_id'], 'alias' );
+				$aliases = array_merge( $aliases, array_fill_keys( $domain_aliases, $domain ) );
+			}
+			set_transient( $cache_key, $aliases );
+		}
+		return $aliases;
+	}
+
+	/**
+	 * Flush and repopulate the domain alias cache.
+	 */
+	public static function update_domain_aliases_cache() {
+		delete_transient( 'switchboard-domain-aliases' );
```

Comment:
It might be useful to store the transient key as a class property since it is referenced in more than one function.